File: easy/338. Counting Bits/338. Counting Bits.py
from typing import List


class Solution:
    def countBits(self, n: int) -> List[int]:
        ans: List[int] = []

        for i in range(n + 1):
            ans.append(self.countBitsInANum(i))

        return ans

    def countBitsInANum(self, num: int) -> int:
        ans = 0

        while num > 0:
            ans += num & 1
            num //= 2

        return ans

File: easy/338. Counting Bits/338. Counting Bits.go
package main

func countBitsInNum(num int) int {
	bits := 0
	for num > 0 {
		bits += num & 1
		num /= 2
	}
	return bits
}

func countBits(n int) []int {
	ans := []int{}
	for i := 0; i <= n; i++ {
		ans = append(ans, countBitsInNum(i))
	}
	return ans
}
